JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/common/MoviesRow.kt

```diff
@@ -119,41 +119,56 @@ fun MoviesRow(
             targetState = movies,
             label = "",
         ) { movieState ->
-            FocusGroup {
-                TvLazyRow(
-                    pivotOffsets = PivotOffsets(parentFraction = 0.07f)
-                ) {
-                    item { Spacer(modifier = Modifier.padding(start = startPadding)) }
+            val focusRequester = remember { FocusRequester() }
+            val movieItemFocusRequester = remember { FocusRequester() }
 
-                    movieState.forEachIndexed { index, movie ->
-                        item {
-                            key(movie.id) {
-                                MoviesRowItem(
-                                    modifier = Modifier.restorableFocus(),
-                                    focusedItemIndex = focusedItemIndex,
-                                    index = index,
-                                    itemWidth = itemWidth,
-                                    itemDirection = itemDirection,
-                                    onMovieClick = onMovieClick,
-                                    movie = movie,
-                                    showItemTitle = showItemTitle,
-                                    showIndexOverImage = showIndexOverImage
-                                )
-                            }
+            TvLazyRow(
+                modifier = Modifier
+                    .focusRequester(focusRequester)
+                    .focusProperties {
+                        enter = {
+                            if (focusRequester.restoreFocusedChild()) FocusRequester.Cancel
+                            else movieItemFocusRequester
                         }
-                        item { Spacer(modifier = Modifier.padding(end = 20.dp)) }
-                    }
+                        exit = {
+                            focusRequester.saveFocusedChild()
+                            FocusRequester.Default
+                        }
+                    },
```

Comment:
Since this part is repeating at a few places, we could abstract this out:

* [Abstracted Modifier](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:tv/integration-tests/playground/src/main/java/androidx/tv/integration/playground/createCustomInitialFocusRestorerModifiers.kt)
* [Helper Utility IfElseModifier](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:tv/integration-tests/playground/src/main/java/androidx/tv/integration/playground/ifElse.kt)
* [Usage](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:tv/integration-tests/playground/src/main/java/androidx/tv/integration/playground/TopNavigation.kt;l=78-92;drc=409d921b5a37ec6857489f327d9cc20141457ab2)

Reply:
Updated to use the utility methods as suggested.